### PR TITLE
new variables

### PR DESCRIPTION
**Pull request by dunn**
`project_base` is what used to be `project_dir`; but if you have a better name for that variable let me know.

### CODE REVIEW

File: roles/alexandria-v2/defaults/main.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 # roles/alexandria-v2/defaults/main.yml
-project_name: alexandria-v2
-capistrano_user: deploy
-capistrano_group: deploy
-minter_state_location: /opt/{{ project_name }}/shared/minter-state
-etd_download_root: /opt/download_root
+project_name: alexadria-v2
```
Comment (acozine) [inline roles/alexandria-v2/defaults/main.yml]: typo here - should be alexandria not alexadria

Comment (dunn) [inline roles/alexandria-v2/defaults/main.yml]: Woops! Fixed, thanks.

**Comment by dunn** [comment]
Rebased.

File: roles/ezid/tasks/main.yml [hunk 2]
```diff
@@ -2,6 +2,5 @@
 # roles/ezid/tasks/main.yml
 - name: add ezid shared config
   become: yes
-  template: src=ezid.yml.j2 dest=/opt/{{ project_name }}/shared/config/ezid.yml owner={{ capistrano_user }} group={{ capistrano_group }} backup=no
-
-      
\ No newline at end of file
+  template: src=ezid.yml.j2 dest={{ config_path }}/ezid.yml \
```
Comment (acozine) [inline roles/ezid/tasks/main.yml:5]: I'd prefer to use Ansible multiline format rather than backslashes for line continuation. You've done this in a number of places, though, so I'll file a ticket for that and we can address those all at once.

File: roles/ffmpeg/defaults/main.yml [hunk 3]
```diff
@@ -1,2 +1,2 @@
 ---
-ffmpeg_path: /opt/install/ffmpeg_sources
\ No newline at end of file
+ffmpeg_path: "{{ project_base }}/install/ffmpeg_sources"
```
Comment (acozine) [inline roles/ffmpeg/defaults/main.yml]: I'd like to see the install_path variable here, so if a user overrides it, the ffmpeg_sources directory will end up in the same location as everything else.

File: roles/hostname/defaults/main.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
```
Comment (acozine) [inline roles/hostname/defaults/main.yml:1]: needs a default for config_path

File: roles/ffmpeg/defaults/main.yml (hunk 3 above)
Comment (dunn) [inline roles/ffmpeg/defaults/main.yml]: Oop, good catch.

File: roles/housekeeping/tasks/users_groups_dirs.yml [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+  group: name={{ deploy_group }} state=present
 
-- name: create cap user
-  user: name={{ capistrano_user }} group={{ capistrano_group }} shell=/bin/bash createhome=yes state=present
+- name: create deploy user
+  user: name={{ deploy_user }} group={{ deploy_group }} shell=/bin/bash createhome=yes state=present
 
-- name: create cap directories
-  file: path={{ item }} state=directory owner={{ capistrano_user }} group={{ capistrano_group }} 
+- name: create project directories
```
Comment (acozine) [inline roles/housekeeping/tasks/users_groups_dirs.yml]: I'd change this from "project directories" to "deployment directories" to make the connection to a deployment tool obvious.

File: roles/housekeeping/tasks/users_groups_dirs.yml [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
+  file: path={{ item }} state=directory owner={{ deploy_user }} group={{ deploy_group }}
   with_items:
-    - /opt/{{ project_name }}/shared/config/initializers
-    - /opt/{{ project_name }}/shared/log
-    - /opt/{{ project_name }}/releases
-    - /home/{{ capistrano_user }}/.ssh
+    - "{{ project_base }}/{{ project_name }}/shared/config/initializers"
```
Comment (acozine) [inline roles/housekeeping/tasks/users_groups_dirs.yml]: What about using config_path here?

File: roles/hydra-stack/defaults/main.yml [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
-install_path: /opt/install
+install_path: "{{ project_base }}/install"
 base_path: prod
-capistrano_user: deploy
-capistrano_group: deploy
+deploy_user: deploy
+deploy_group: deploy
 
 fedora_ver: 4.5.0
 fedora_256: f9c778c4285dcbb7ab228f43e5e30ea7511a5553ca52a34e274e1e390cb5187e
```
Comment (acozine) [inline roles/hydra-stack/defaults/main.yml:9]: Needs default value for project_base.

File: roles/imagemagick/defaults/main.yml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 ---
 # roles/imagemagick/defaults/main.yml
```
Comment (acozine) [inline roles/imagemagick/defaults/main.yml]: default for project_base

File: roles/ldap/defaults/main.yml [hunk 9]
```diff
@@ -1,8 +1,8 @@
 ---
 # roles/ldap/defaults/main.yml
```
Comment (acozine) [inline roles/ldap/defaults/main.yml:2]: default for config_path

File: roles/marmotta/defaults/main.yml [hunk 10]
```diff
@@ -5,7 +5,7 @@ tomcat: /usr/share/tomcat
 
 marmotta_ver: 3.3.0
 marmotta_256: 34c37281b6875ea95da391ae5bbe059f31091f3c6bf2529539241b35690c84b0
-marmotta_home: /opt/marmotta
+marmotta_home: "{{ project_base }}/marmotta"
```
Comment (acozine) [inline roles/marmotta/defaults/main.yml:9]: default value for project_base in roles/marmotta/defaults/main.yml

File: roles/ruby/defaults/main.yml [hunk 11]
```diff
@@ -3,4 +3,4 @@ ruby_ver: 2.2.3
 ruby_abi: 2.2
 ruby_256: df795f2f99860745a416092a4004b016ccf77e8b82dec956b120f18bdc71edce
 # rails_ver: 4.2.0
-install_path: /opt/install
+install_path: "{{ project_base }}/install"
```
Comment (acozine) [inline roles/ruby/defaults/main.yml:7]: default for project_base here too

File: roles/services/defaults/main.yml [hunk 12]
```diff
@@ -1,7 +1,7 @@
 ---
-install_path: /opt/install
-capistrano_user: deploy
-capistrano_group: deploy
+install_path: "{{ project_base }}/install"
```
Comment (acozine) [inline roles/services/defaults/main.yml:3]: and here

File: roles/smtp/defaults/main.yml [hunk 13]
```diff
@@ -1,8 +1,8 @@
 ---
 # roles/smtp/defaults/main.yml
```
Comment (acozine) [inline roles/smtp/defaults/main.yml:2]: config_path default here

File: roles/vagrant/tasks/vagrant_housekeeping.yml [hunk 14]
```diff
@@ -1,15 +1,15 @@
 ---
 - name: temporarily move Guest Additions
   become: yes
-  shell: mv "/opt/VBoxGuestAdditions"* /tmp
+  shell: mv "{{ project_base }}/VBoxGuestAdditions"* /tmp
```
Comment (acozine) [inline roles/vagrant/tasks/vagrant_housekeeping.yml:4]: default value for project_base in the vagrant/defaults/main.yml

File: roles/ezid/tasks/main.yml (hunk 2 above)
Comment (acozine) [inline roles/ezid/tasks/main.yml:5]: Added #62

**Comment by dunn** [comment]
I shouldn't have mixed up `config_path` with the `capistrano_user` rename.  Separated those commits.

File: roles/housekeeping/tasks/users_groups_dirs.yml (hunk 5 above)
Comment (dunn) [inline roles/housekeeping/tasks/users_groups_dirs.yml]: Good idea; done.

**Comment by dunn** [comment]
Think I added all the default values, thanks for catching that.

File: roles/alexandria-v2/tasks/adrl-dm.yml [hunk 15]
```diff
@@ -1,18 +1,18 @@
...
-           chdir=/opt/ingest/metadata/adrl-dm
+           chdir={{ project_base }}/ingest/metadata/adrl-dm
 
 - name: ensure {{ deploy_user }} owns ingest directories
   file: path={{ item }} owner={{ deploy_user }} group={{ deploy_group }} \
         state=directory recurse=yes
   with_items:
-    - /opt/download_root
-    - /opt/ingest/metadata
+    - "{{ project_base }}/download_root"
```
Comment (acozine) [inline roles/alexandria-v2/tasks/adrl-dm.yml]: If you use the etd_download_root variable here, then any override to that variable will change both the directory and the config pointing to it.

Comment (dunn) [inline roles/alexandria-v2/tasks/adrl-dm.yml]: :+1:

File: roles/app-config/defaults/main.yml [hunk 16]
```diff
@@ -1,7 +1,7 @@
 ---
 # roles/app-config/defaults/main.yml
 project_name: sufia-project
```
Comment (acozine) [inline roles/app-config/defaults/main.yml:3]: I missed this one - the app-config role also needs a default for project_base

Comment (dunn) [inline roles/app-config/defaults/main.yml:3]: Ah, yeah.

File: roles/alexandria-v2/defaults/main.yml [hunk 17]
```diff
@@ -1,7 +1,7 @@
 ---
 # roles/alexandria-v2/defaults/main.yml
 project_name: alexandria-v2
```
Comment (acozine) [inline roles/alexandria-v2/defaults/main.yml]: and project_base default here - please match the default value to the rest of the role defaults

File: roles/ec2/tasks/ec2_housekeeping.yml [hunk 18]
```diff
@@ -1,9 +1,9 @@
 ---
-- name: format volume for /opt
+- name: format volume for {{ project_base }}
```
Comment (acozine) [inline roles/ec2/tasks/ec2_housekeeping.yml:2]: and also in the ec2 role

File: roles/generic-centos/tasks/generic_setup.yml [hunk 19]
```diff
@@ -1,7 +1,7 @@
 ---
-- name: format volume for /opt
+- name: format volume for {{ project_base }}
```
Comment (acozine) [inline roles/generic-centos/tasks/generic_setup.yml:2]: and also here

File: roles/housekeeping/defaults/main.yml [hunk 20]
```diff
@@ -1,8 +1,9 @@
 ---
-capistrano_user: deploy
-capistrano_group: deploy
-capistrano_keys:
+deploy_user: deploy
```
Comment (acozine) [inline roles/housekeeping/defaults/main.yml:2]: config_path default here?

File: roles/alexandria-v2/defaults/main.yml (hunk 17 above)
Comment (dunn) [inline roles/alexandria-v2/defaults/main.yml]: Done in a separate commit, since it was introduced before this PR.

File: roles/housekeeping/defaults/main.yml (hunk 20 above)
Comment (dunn) [inline roles/housekeeping/defaults/main.yml:2]: Yep yep.

File: roles/ldap/defaults/main.yml [hunk 21]
```diff
@@ -1,8 +1,9 @@
...
 project_name: sufia-project
-capistrano_user: deploy
-capistrano_group: deploy
+deploy_user: deploy
+deploy_group: deploy
+config_path: "{{ project_base }}/{{ project_name }}/shared/config"
```
Comment (acozine) [inline roles/ldap/defaults/main.yml]: and a default for project_base here, otherwise the config_path default won't function

File: roles/housekeeping/defaults/main.yml (hunk 20 above)
Comment (dunn) [inline roles/housekeeping/defaults/main.yml]: :see_no_evil:

File: roles/ldap/defaults/main.yml (hunk 21 above)
Comment (dunn) [inline roles/ldap/defaults/main.yml]: :see_no_evil:

**Comment by dunn** [comment]
Caught a few more in hostname and smtp.

**Comment by acozine** [comment]
Okay, I'll merge the PR then test with no overrides and see what, if anything, breaks.
